src/Factory/HalConfigFactory.php

```diff
@@ -6,20 +6,28 @@
 
 namespace ZF\Hal\Factory;
 
-use Zend\ServiceManager\FactoryInterface;
-use Zend\ServiceManager\ServiceLocatorInterface;
+use Interop\Container\ContainerInterface;
+use Zend\ServiceManager\Factory\FactoryInterface;
```

Comment:
This implementation should be done in a way that is backwards compatible with v2 releases. As such, you can implement `Zend\ServiceManager\FactoryInterface`, and provide both `__invoke()` (v3 signature) and `createService()` (v2 signature) methods.

Alternately, don't implement the interface, and implement only `__invoke()`, as invokable classes may be used as factories in both versions. If you do this, however, implement `createService()` as well, and have it proxy:

```php
public function createService($container)
{
    return $this($container);
}
```

(I've done that on the develop branch already for a few factories; it simplifies the code, but you lose some type-safety that you would retain with the other approach.)